source/language/types.rst

```diff
@@ -12,28 +12,66 @@ Variable identifiers must begin with a letter [A-Za-z], an underscore, a
 percent sign, or an element from the Unicode character categories
 Lu/Ll/Lt/Lm/Lo/NI :cite:`noauthorUnicodeNodate`.
 Continuation characters may contain numbers. Variable identifiers may
-not override a reserved identifier. All qubits are global variables.
+not override a reserved identifier.
+
+All qubits are global variables.
+
 When qubits are declared, their state is initially undefined. Qubits
 cannot be declared within gates or subroutines. This simplifies OpenQASM
 significantly since there is no need for quantum memory management.
-However, it also means that users or compiler have to explicitly manage
-the quantum memory. In addition to being assigned values within a
-program, all of the classical types can be initialized on declaration.
+However, it also means that users or compilers have to explicitly manage
+the quantum memory.
+
+OpenQASM supports all scalar classical types as defined in the C99 Standard:
+http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf
+
+Additionally, OpenQASM 3.0 supports compile-time fixed-size C-style bit
+arrays.  Variable-Length Arrays (VLA's) are not supported in OpenQASM 3.0.
+
+Missing - Needs Clarification:
+------------------------------
+Does an Array automatically decay to the pointer of its first element, as in C?
+-------------------------------------------------------------------------------
+Can an Array be passed as argument to a function or subroutine, as in C?
+------------------------------------------------------------------------
+Signed / Unsigned
+-----------------
+Overflow / Underflow
+--------------------
+Wrap-Around (unsigned)
```

Comment:
Good point. At least with the `angle[n]` type we explicitly want wrap-around throwing an errors. For consistency sake, we might want to adopt this behavior everywhere.